cli/verifier/versions_match.go

```diff
@@ -0,0 +1,84 @@
+// Copyright 2025 Edgeless Systems GmbH
+// SPDX-License-Identifier: BUSL-1.1
+
+package verifier
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+
+	"github.com/edgelesssys/contrast/internal/kuberesource"
+	"github.com/regclient/regclient/types/ref"
+
+	applycorev1 "k8s.io/client-go/applyconfigurations/core/v1"
+	applymetav1 "k8s.io/client-go/applyconfigurations/meta/v1"
+)
+
+var versionedResources = []identifier{
+	{name: "coordinator", podRole: "coordinator"},
+	{name: "contrast-initializer"},
+	{name: "contrast-service-mesh"},
+}
+
+// VersionsMatch verifies that the cli version matches the version of the used resources.
+type VersionsMatch struct {
+	Version string
+}
+
+// Verify verifies that the cli version matches the version of the used resources.
+func (v *VersionsMatch) Verify(toVerify any) error {
+	// ignore this verifier for preview and dev builds.
+	if strings.Contains(v.Version, "dev") || strings.Contains(v.Version, "pre") {
+		return nil
+	}
+
+	var findings error
+
+	kuberesource.MapPodSpecWithMeta(toVerify, func(
+		meta *applymetav1.ObjectMetaApplyConfiguration,
+		spec *applycorev1.PodSpecApplyConfiguration,
+	) (*applymetav1.ObjectMetaApplyConfiguration, *applycorev1.PodSpecApplyConfiguration) {
+		if spec.RuntimeClassName == nil || !strings.HasPrefix(*spec.RuntimeClassName, "contrast-cc") {
+			return meta, spec
+		}
+
+		for _, container := range spec.Containers {
+			if !matchesResource(*container.Name, meta.Annotations["contrast.edgeless.systems/pod-role"]) {
+				continue
+			}
+
+			r, err := ref.New(*container.Image)
```

Comment:
Nit: this will panic if the image is unset. We should report a proper error back, since this is never going to work on any k8s cluster.

Reply:
Mh, that's true, but this doesn't really seem like the right verifier to handle this.I have reworked the `ImageRefValid` verifier since this seems to fall within its domain, and reordered the verifiers to ensure that one runs first. As a precaution, let's additionally just skip empty images in the `VersionsMatch` verifier.

Is this OK with you?

Reply:
Yup, sgtm